From 2cb6df61f35a7a80209fd873d7f1113040c9afb8 Mon Sep 17 00:00:00 2001
From: Tristan Cacqueray <tdecacqu@redhat.com>
Date: Wed, 20 Jan 2016 12:37:42 -0500
Subject: [PATCH] Set logserver variables using temp url key only

When Send-Temp-Url-Key is set to False and X-Account-Meta-Temp-Url-Key
is defined, there is no need for a swiftclient connection and the
zuul_swift_upload.py works using the provided temp_url_key.

Without this, the [swift] configuration section always requires OS_
settings, even if temp_url_key and logserver_prefix are already defined.

Change-Id: I8990359d47e9d5065d34af3faea2d24e63e736b8
---

diff --git a/zuul/launcher/gearman.py b/zuul/launcher/gearman.py
index 69fb71b..437fc45 100644
--- a/zuul/launcher/gearman.py
+++ b/zuul/launcher/gearman.py
@@ -224,7 +224,10 @@
         # NOTE(jhesketh): The params need to stay in a key=value data pair
         # as workers cannot necessarily handle lists.
 
-        if job.swift and self.swift.connection:
+        if job.swift and (self.swift.connection or (
+            self.config.has_option('swift', 'X-Account-Meta-Temp-Url-Key') and
+            self.config.has_option('swift', 'Send-Temp-Url-Key') and
+            self.config.getboolean('swift', 'Send-Temp-Url-Key') == False)):
 
             for name, s in job.swift.items():
                 swift_instructions = {}
